tket/src/Predicates/PassGenerators.cpp

```diff
@@ -480,14 +481,28 @@ PassPtr gen_cx_mapping_pass(
     bool delay_measures) {
   OpTypeSet gate_set = all_single_qubit_types();
   gate_set.insert(OpType::CX);
-  PassPtr rebase_pass =
-      gen_rebase_pass(gate_set, CircPool::CX(), CircPool::tk1_to_tk1);
+  PassPtr rebase_pass = gen_auto_rebase_pass(gate_set);
   PassPtr return_pass =
       rebase_pass >> gen_full_mapping_pass(arc, placement_ptr, config);
   if (delay_measures) return_pass = return_pass >> DelayMeasures();
   return_pass = return_pass >> rebase_pass >>
                 gen_decompose_routing_gates_to_cxs_pass(arc, directed_cx);
-  return return_pass;
+  Transform t{[=](Circuit& circ) {
```

Comment:
I am not totally confident in this: should we be passing a `std::shared_ptr<unit_bimaps_t>` as well as a `Circuit&` to the `Transform` constructor? If so what should we do with it?

Reply:
What would the `unit_bimaps_t` object be for? I might be forgetting some TKET infrastructure here. Certainly we don't need to pass any kind of qubit placement if that's what it's referring to (as the circuit UnitID are updated).

Reply:
Wasn't sure myself. Seems to be used in placement passes where units are renamed (e.g. `gen_placement_pass()`). So I assume not necessary here.

Reply:
Please check my last commit. I don't know if this is correct, but I don't know what else to do with the maps either!

Reply:
```  Transform t{[=](Circuit& circ, std::shared_ptr<unit_bimaps_t> maps) {

      // Relabel all two-qubit gates
      CompilationUnit cu(circ);
      bool changed = rebase_pass->apply(cu);
      circ = cu.get_circ_ref();
      // Now try placement, falling back on "LinePlacement" if "GraphPlacement" fails
      try {
        changed = changed | placement_ptr->place(circ, maps);
      } catch (const std::runtime_error& e) {
        std::stringstream ss;
        ss << "PlacementPass failed with message: " << e.what()
          << " Fall back to LinePlacement.";
        tket_log()->warn(ss.str());
        Placement::Ptr line_placement_ptr = std::make_shared<LinePlacement>(
            placement_ptr->get_architecture_ref());
        changed = changed | line_placement_ptr->place(circ, maps);

      }
      // Now route
      MappingManager mm(std::make_shared<Architecture>(arc));
      changed = changed | mm.route_circuit_with_maps(circ, config, maps);
      
      // Now decompose routing gates to cx gates (this won't change maps)
      cu(circ);
      if(delay_measures){
        DelayMeasures()->apply(cu);
      }
      gen_decompose_routing_gates_to_cxs_pass(arc, directed_cx)->apply(cu);
      circ = cu.get_circ_ref();
      return changed;
  }```